image-rs/src/pull.rs

```diff
@@ -142,10 +142,16 @@ impl<'a> PullClient<'a> {
         };
 
         let decryptor = Decryptor::from_media_type(&layer.media_type);
+        let decryptor_i = decryptor.clone();
```

Comment:
not sure about this naming convention.

I've seen people create a new scope around their closure and then shadow the variables like this.
```
tokio::spawn({
    let foo = foo.clone();
    async move {
        blah(foo)
    }
});
```
not sure if that would work here. If not maybe just `decryptor_clone`.

Reply:
Oh I test locally like 
```rust
// decrypt_config is `Option<&str>`, so we need to clone the &str to string outside the closure
let decrypt_config_clone = decrypt_config.as_ref().map(|inner| inner.to_string());
let decrypt_key = tokio::task::spawn_blocking(move || {
                let decryptor_clone = decryptor.clone();
                let layer_clone = layer.clone();
                decryptor_clone
                    .get_decrypt_key(&layer_clone, &decrypt_config_clone.as_deref())
                    .map_err(|e| anyhow!("failed to get decrypt key {}", e.to_string()))
            })
```

The compiler still cries because the moved value, hinting that we must do `clone` operation outside the closure.

Reply:
Let me just rename the variables

Reply:
The  new scope has to go before the closure not inside of it.

Reply:
Oh! I get your point now. Let me rewrite as
```rust
let decrypt_key = tokio::task::spawn_blocking({
                let decryptor_clone = decryptor.clone();
                let layer_clone = layer.clone();
                let decrypt_config_clone = decrypt_config.as_ref().map(|inner| inner.to_string());
                move || {
                    decryptor_clone
                        .get_decrypt_key(&layer_clone, &decrypt_config_clone.as_deref())
                        .context("failed to get decrypt key")
                }
            })
```

Reply:
If you have this new scope you can shadow the variables (i.e. `let decryptor = decryptor.clone()`), but it might be clearer to have new names. Up to you.

Reply:
Good, changed the name.